src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -318,6 +326,7 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
             market_address
         );
         let swapper_address = signer::address_of(swapper);
+        assert!(account::exists_at(swapper_address), E_SWAPPER_DOES_NOT_EXIST);
```

Comment:
@alnoki so I added this here because on line `330` previously, the function call fails if the account already exists, which it will the vast majority of the time. I added an error code `E_SWAPPER_DOES_NOT_EXIST` and a simple test for it.

The error is thrown only in `swap` and just prior to simulating the swap. We could also put it inside the swap simulation function but that's your call.

Perhaps it's overkill, but I think it's the cleanest way to handle it.

Reply:
```suggestion
        account::create_account_if_does_not_exist(swapper_address);
        coin::register<AptosCoin>(swapper);
```

Reply:
Assuming you're fine with this suggestion:

- [ ] Remove the error code since it is no longer needed

Reply:
Okay I figured we'd probably want to avoid doing the error code, this would be great but I think `create_account_if_does_not_exist` is private unfortunately :( we can just add our own inline for it?

Reply:
We resolved this in #24